benches/bench_basic.rs

```diff
@@ -119,35 +119,29 @@ fn bench_long_pipeline(b: &mut Bencher) {
 fn bench_async_long_pipeline(b: &mut Bencher) {
     let client = get_client();
     let mut runtime = Runtime::new().unwrap();
-    let mut con = Some(runtime.block_on(client.get_async_connection()).unwrap());
+    let mut con = runtime.block_on(client.get_async_connection()).unwrap();
 
     let pipe = long_pipeline();
 
     b.iter(|| {
-        con = runtime
-            .block_on(future::lazy(|| {
-                pipe.clone()
-                    .query_async(con.take().expect("Connection"))
-                    .map(|(con, ())| Some(con))
-            }))
+        runtime
+            .block_on(async { pipe.query_async(&mut con).map_ok(|()| ()).await })
```

Comment:
What is this `map_ok()` doing? It reads like a no-op.

Reply:
Just a remnant from futures-0.1 I tend to shut of my brain while refactoring :)